model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/userSession/client/entity/JpaClientSessionEntity.java

```diff
@@ -0,0 +1,319 @@
+        return userSessionId == null ? null : userSessionId.toString();
+    }
+
+    @Override
+    public void setUserSessionId(String userSessionId) {
+        this.userSessionId = UUID.fromString(userSessionId);
+    }
+
+    @Override
+    public String getRealmId() {
+        return metadata.getRealmId();
+    }
+
+    @Override
+    public void setRealmId(String realmId) {
+        metadata.setRealmId(realmId);
+    }
+
+    @Override
+    public String getClientId() {
+        return metadata.getClientId();
+    }
+
+    @Override
+    public void setClientId(String clientId) {
+        metadata.setClientId(clientId);
+    }
+
+    @Override
+    public String getAuthMethod() {
+        return metadata.getAuthMethod();
+    }
+
+    @Override
+    public void setAuthMethod(String authMethod) {
+        metadata.setAuthMethod(authMethod);
+    }
+
+    @Override
+    public String getRedirectUri() {
+        return metadata.getRedirectUri();
+    }
+
+    @Override
+    public void setRedirectUri(String redirectUri) {
+        metadata.setRedirectUri(redirectUri);
+    }
+
+    @Override
+    public Long getTimestamp() {
+        if (isMetadataInitialized()) return metadata.getTimestamp();
+        return timestamp;
+    }
+
+    @Override
+    public void setTimestamp(Long timestamp) {
+        metadata.setTimestamp(timestamp);
+    }
+
+    @Override
+    public Long getExpiration() {
+        if (isMetadataInitialized()) return metadata.getExpiration();
+        return expiration;
+    }
+
+    @Override
+    public void setExpiration(Long expiration) {
+        metadata.setExpiration(expiration);
+    }
+
+    @Override
+    public String getAction() {
+        if (isMetadataInitialized()) return metadata.getAction();
+        return action;
+    }
+
+    @Override
+    public void setAction(String action) {
+        metadata.setAction(action);
+    }
+
+    @Override
+    public String getCurrentRefreshToken() {
+        return metadata.getCurrentRefreshToken();
+    }
+
+    @Override
+    public void setCurrentRefreshToken(String currentRefreshToken) {
+        metadata.setCurrentRefreshToken(currentRefreshToken);
+    }
+
+    @Override
+    public Integer getCurrentRefreshTokenUseCount() {
+        return metadata.getCurrentRefreshTokenUseCount();
+    }
+
+    @Override
+    public void setCurrentRefreshTokenUseCount(Integer currentRefreshTokenUseCount) {
+        metadata.setCurrentRefreshTokenUseCount(currentRefreshTokenUseCount);
+    }
+
+    @Override
+    public Boolean isOffline() {
+        if (isMetadataInitialized()) return metadata.isOffline();
+        return offline;
+    }
+
+    @Override
+    public void setOffline(Boolean offline) {
+        metadata.setOffline(offline);
+    }
+
+    @Override
+    public Map<String, String> getNotes() {
+        return Collections.unmodifiableMap(notes.stream().collect(Collectors.toMap(JpaClientSessionNoteEntity::getName, JpaClientSessionNoteEntity::getValue)));
```

Comment:
Is it really necessary to wrap this in an unmodifiable map?

Reply:
No, it was copied from `JpaPolicyEntity.getConfig()` method. I'll update it.